libs/datamodel/core/src/diagnostics/collection.rs

```diff
@@ -7,32 +7,28 @@ use crate::diagnostics::warning::DatamodelWarning;
 /// It is used to not error out early and instead show multiple errors at once.
 #[derive(Debug, Clone)]
 pub struct Diagnostics {
+    /// TODO: these two fields are public for tests. We should strive towards
```

Comment:
What about `#[mod(test)]`

Reply:
Or getters?

Reply:
`#[cfg(test)]` won't work because the tests are integration tests. Getters we can do, yes, I'll do that.

Reply:
Done :+1:  Pushing this soon.